streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -349,6 +343,7 @@ public void handleAssignment(final Map<TaskId, Set<TopicPartition>> activeTasks,
             throw first.getValue();
         }
 
+        taskExecutor.registerTaskErrors(topologyMetadata.applicationErrorTasks());
```

Comment:
We have two callers to register task errors, one is inside the task executor when we hit an exception and then immediately populate the map, which makes sense; and the other one is here that we register from the topology metadata, but the metadata's registered map is from the executor as well. So it seems to me that we have a circular call trace that populates of the error mappings, and not clear why.. Could you explain a bit?